src/viewer.js

```diff
@@ -2602,8 +2621,26 @@ function onCanvasKeyPress( event ) {
                   this.viewport.applyConstraints();
                 }
                 return false;
+            case 114: //r
+            case 82: //R
+              if(this.flipped){
+                this.viewport.setRotation(this.viewport.degrees - 90);
+              } else{
+                this.viewport.setRotation(this.viewport.degrees + 90);
+              }
+              this.viewport.applyConstraints();
+              return false;
+            case 70: //F
+            case 102: //f
+              this.flipped = !this.flipped;
```

Comment:
Looks like this might as well just call `doFlip`.

Reply:
Done.

Reply:
@iangilman , you meant to say that line 2635 shouldn't be there, just call a single function toggleFlip which toggles the viewer and navigator flip state at the same time, correct?
This line is there because we are flipping only the viewer. Then, if a navigator exists, it should be flipped too.

Reply:
If I understood it correctly, You want a new function anywhere which contains this code.

like: 
case 102: //f
              toggleFlip();
              return false;
(...)

function toggleFlip(viewer){
              viewer.flipped = !viewer.flipped;
              if(viewer.navigator){
                viewer.navigator.toggleFlip();
              }
              viewer._forceRedraw = !viewer._forceRedraw;
              viewer.forceRedraw();
}

Reply:
Sorry, now I understood. I'will change it.

Reply:
onFlip is not a viewer function, is a openseadragon method. I would have to create a new function for the viewer. I think, I will try to save flip state on viewport and make this change too.